packages/expo-keep-awake/android/src/main/java/expo/modules/keepawake/ExpoKeepAwakeManager.java

```diff
@@ -0,0 +1,70 @@
+package expo.modules.keepawake;
+
+import android.app.Activity;
+import android.view.WindowManager;
+
+import org.unimodules.core.ModuleRegistry;
+import org.unimodules.core.errors.CurrentActivityNotFoundException;
+import org.unimodules.core.interfaces.ActivityProvider;
+import org.unimodules.core.interfaces.InternalModule;
+import org.unimodules.core.interfaces.ModuleRegistryConsumer;
+import org.unimodules.core.interfaces.services.KeepAwakeManager;
+
+import java.util.Collections;
+import java.util.List;
+
+public class ExpoKeepAwakeManager implements KeepAwakeManager, InternalModule, ModuleRegistryConsumer {
+
+  private ModuleRegistry mModuleRegistry;
+  private boolean mIsActivated;
+
+  @Override
+  public void setModuleRegistry(ModuleRegistry moduleRegistry) {
+    this.mModuleRegistry = moduleRegistry;
+  }
+
+  private Activity getCurrentActivity() throws CurrentActivityNotFoundException {
+    ActivityProvider activityProvider = mModuleRegistry.getModule(ActivityProvider.class);
+    if (activityProvider.getCurrentActivity() != null) {
+      return activityProvider.getCurrentActivity();
+    } else {
+      throw new CurrentActivityNotFoundException();
+    }
+  }
+
+  @Override
+  public void activate(final Runnable done) throws CurrentActivityNotFoundException {
+    final Activity activity = getCurrentActivity();
+
+    if (activity != null) {
+      activity.runOnUiThread(() -> {
+        activity.getWindow().addFlags(WindowManager.LayoutParams.FLAG_KEEP_SCREEN_ON);
+        mIsActivated = true;
+        done.run();
```

Comment:
Is running `done.run()` on UI thread safe? idk, just a loud thought 🤔

Reply:
Honestly, I analyzed it only from JS perspective, and it seems safe then. We will only resolve Java promise in UI Thread, but actual JS Promise will get resolved on the other side of the JSBridge, so shrug.

But it looks like it might cause troubles in Android code, if someone decides to perform some logic on that callback... I will move that out of this thread then.